### ISSUE TITLE
Why use `Timex.Duration` instead of `Time`?

### ISSUE THREAD

**Issue by martinmaillard** [bitwalker/timex_ecto#37]
I was wondering: what made you chose `Timex.Duration` for `Timex.Ecto.Time` fields instead of the `Time` type from standard lib?

**Comment by bitwalker** [comment]
Two reasons, first, it is the direct equivalent of the old `Timex.Time` type (just an Erlang timestamp), where `Time` is not (it can only be used to represent a 24 hour clock). The `Timex.Time` API was replaced with `Timex.Duration`, along with the new struct. However I have not yet added a complete `Timex.Time` API to work with the new `Time` type, mostly because there hasn't been much need beyond the one function pair `to_12hour_clock/1` and `to_24hour_clock/1`. Because of this, I felt that getting a `Timex.Duration` type back would be far more useful, both for the above reasons, plus it allows non-pgsql databases to use time to represent intervals/durations as well.

In 3.0.3 I added `Duration.to_time/1`, `Duration.to_time!/1`, and `Duration.from_time/1` for converting `Duration` back and forth between `Time`, so if you desire a `Time` struct, it is easy to obtain one from a `Duration`.

I'm open to changing this at some point, but I was mostly looking for a relatively one-to-one migration from the old timex to the new, without introducing too many changes.
